### DISCUSSION TITLE
iOS message notifications not delivered via FCM - firebase_messaging 9.x.x

### DISCUSSION THREAD

**Discussion by tkeithblack** [firebase/flutterfire#5122] [Q&A]
I recently upgraded to firebase_messaging 9.0.0 because it indicated that there was new support for iOS push notifications. However, it appears that only messages sent through firebase are processed on iOS.

Is there any plan to deliver message notifications sent to iOS from severs other than the firebase FCM method?

I see it is possible to display foreground notifications (heads-up), however, I need my app to process the messages the same as when running in Android (i.e. using onMessage, onMessageOpenedApp, onBackgroundMessage). 

Messages processed by my app do not come through gcm.message service.

Thanks,
Keith

**Comment by miqueasgutierrez** [comment]
Hello, Test with the INDIGITALL Service, it is also Free for 30000 Devices, it allows to send Push Notifications with also geo-located images, programmed among many more functions, here is the link of the documentation to install it easily: https: //docs.indigitall .you eat/
